RRF-117: Sort Picker

arianaantonio opened the pull request:
- Added sort and filter buttons
- Added sort picker and options to sort
- Sorting with API
- Added footer buttons and functionality for pagination
- Modified constraints
- Updated sort options

arianaantonio committed:
RRF-117: Sort Picker
- Added sort and filter buttons
- Added sort picker and options to sort
- Sorting with API
- Added footer buttons and functionality for pagination

arianaantonio committed:
Merge branch 'master' into feature/RRF-117-Sort-Picker

arianaantonio committed:
RRF-117: Sort Picker
- Modified constraints
- Updated sort options

KingOfBrian commented on Demo/RCHSearchViewController.swift:
```diff
@@ -116,16 +193,51 @@ extension RCHSearchViewController: UISearchBarDelegate, UICollectionViewDelegate
+            break
+        case .relevance:
+            sortString = "product_relevance"
+        }
```
I'd add this stuff as an extension to `SortingOptions` to thin this out

KingOfBrian commented on Demo/RCHSearchViewController.swift:
```diff
@@ -86,14 +127,51 @@ extension RCHSearchViewController: UISearchBarDelegate, UICollectionViewDelegate
     func dismissSearch() {
         view.endEditing(true)
         autocompleteSuggestions.removeAll()
+        sortButton.setTitle("Sort", for: .normal)
```
Is there any way to keep the sort state around between searches easily?

KingOfBrian commented on Demo/RCHSearchViewController.swift:
```diff
@@ -86,14 +127,51 @@ extension RCHSearchViewController: UISearchBarDelegate, UICollectionViewDelegate
+        pickerView.isHidden = true
+        
+        previousButton.isEnabled = false
+        nextButton.isEnabled = false
```
There's a bug with tapping on the search field when the picker is open. If you use the responder chain to present the picker view, this should clean up a bit. Instead of adding `pickerView` and `toolbar` directly to the view hierarchy, you can return them for the properties `inputView` and `inputAccessoryView`. Just have this view controller return `true` for `canBecomeFirstResponder`, and to present the picker call `self.becomeFirstResponder()`. I think it will work a bit more naturally, and will cover the tab bar and will automatically dismiss if the search field is pressed.

KingOfBrian commented on Demo/RCHSearchViewController.swift:
```diff
@@ -13,22 +13,57 @@ private let reuseIdentifier = "productCell"
+        didSet {
+           searchForProducts()
+        }
+    }
```
I'm not sure I get the difference between pageCount and pageStart. pageStart is duplicated state that could be computed with `pageCount / 20`.

arianaantonio committed:
RRF-117: Sort Picker
- Extended SortingOptions enum
- Moved api calls out of didSets
- Removed pageStart

arianaantonio committed:
RRF-116: Filter Modal
- Added filter ActionSheet

arianaantonio committed:
RRF-116: Filter modal
- Made UIAlertController extension

KingOfBrian committed:
Merge pull request #14 from Raizlabs/feature/RRF-116-Filter-Modal

RRF-116: Filter Modal
